packages/gatsby/src/utils/webpack.config.js

```diff
@@ -423,6 +414,28 @@ module.exports = async (
       ],
     }
 
+    // if react-hot-reloading
+    if (
+      stage === `develop` &&
+      process.env.GATSBY_HOT_LOADER !== `fast-refresh`
+    ) {
+      resolve.alias[`react-hot-loader`] = path.dirname(
+        require.resolve(`react-hot-loader/package.json`)
+      )
+
+      // We need to add @hot-loader/react-dom to support hot-loader work with hooks
+      try {
+        resolve.alias[`react-dom/server$`] = require.resolve(`react-dom/server`)
+        resolve.alias[`react-dom`] = require.resolve(`@hot-loader/react-dom`)
```

Comment:
Should we add `@hot-loader/react-dom` to `peerDependencies`? Wonder if this would cause troubles with PnP if we don't?

Reply:
i'm not sure, I can do a test run with yarn 2. Maybe optionalDependencies so we don't get npm warnings.

Reply:
added it to peerdeps :) becasue optionalDeps get installed. Tested on yarn2 and not complaints